### DISCUSSION TITLE
Feature Request : Add progress note to task and be able to put the active task on hold, start a new one and resume the previous task

### DISCUSSION THREAD

**Discussion by mochouinard** [kimai/kimai#3814] [General]
I tried in the past to do a patch, but I failed to have enough time to implement this (and I would probably change stuff in way that are not welcomed back).

So the first one is so I can track quickly progress within my work.  This could be used for multiple things, like having a team leader be able to check the progress of the work.  My goal is so I can remember the steps I took to find the resolutions, so people who review them (mostly myself), can understand what happen and what needed to be fixed later on.
It could be a simple button that bring a popup which would append a message in the description with the time/date of that note.  Not very future proof but it could work.  Extra feature request : Maybe also have the option that when you click end on an active task, it bring a popup with the ability to add a note.

The second one is more complicated, and could be linked with the first one if designed this way :  Be able to have multiple time entry per task.  This would allow to put a task on hold, and resume it latter on (with or without comment).
It occur so often that I start working on something that might take me 4 hour to do, but during that time, I get 5,6 different phone call form different people, which I need to stop my current task and start a new one.  And after that 5 min call, I need to close it, and I can't really start the previous back since it just have 1 start & end time, so I would create a new whole entry.

One 'easy' way to do it, would be to have an array of start/stop time, and calculate the total time based on this.  When multiple entry exist, have it display a table of the start / stop time so it can be manually edited.  I have no problem that this is only 24h based from the first start time.  So if I start to work again the next day on a previous ticket, well I can't and I need to start a new one (It could be 'nice' if we could link them with an ID, but I REALLY don't care about this)

I guess a third feature request would be to have title to a task. Customer, Project and activity is nice, but when you glance into a list, it doesn't give much details of what happen during the day.

So I can still work on it, but I would need some guidance in how you would like me to try to implement these features.

Thanks.

**Comment by kevinpapst** [comment]
Sounds like https://www.kimai.org/store/task-management-bundle.html

**Reply by mochouinard** [reply]
I don't think so looking at the demo page.

I used the word task for lack of other word to describe what is a 'time entry'.  I don't care about scheduling things for the future or assigning task to people.

I just want to press the Start button then I start working on something (can be a client that call me out of the blue to ask me things), or an email just came in I need to work on.  So those are not planned before hand, and it need to be quick to be worth doing(else I just stop tracking my time all together).

I want it to be quick since I don't want to waste my time and the client time to build a 'task' when they call me.  Which is why also updating the 'task' at the end with what happen would be useful since when I start the timer, I might not even know what the client want yet.  That said, I would normally fill it in with the client info and quick description the client gave me(which would be more a title in my case), to begin with, but to be sure the timer run, I just save right away to not lose it.  Once the call is done, I would like to be able to edit it with a resolution description and maybe fix the title if I didn't understand the request.

But again, to be quick, I would want to switch to another 'task' if I receive another call while I'm taking care of an email for example.
